File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteContractsOption3/QuoteContractOption3/QuoteContractOption3Validator.vb
Namespace Entity

Namespace QuoteContractOption3s

Public Class QuoteContractOption3Validator
    'make sure that contact object is valid
    Inherits BaseValidator

    Public Sub Validate(ByVal oQuoteContractOption3 As QuoteContractOption3)

        'make sure that contact object is valid
        If oQuoteContractOption3.Errors.Count > 0 Then
            Dim de As DictionaryEntry
            For Each de In oQuoteContractOption3.Errors
                Me.AddCriticalMessage(CStr(de.Value))
            Next
        End If

                If oQuoteContractOption3.QuoteContractReference.Trim.Length = 0 Then
                    AddCriticalMessage("* Quote Contract Referenece is required")
                End If

                If InStr(oQuoteContractOption3.QuoteContractReference, "-") > 0 Then
                    AddCriticalMessage("* Quote Contract Referenece cannot contain '-'.")
                End If

                If oQuoteContractOption3.QuoteContractStatusID = 0 Then
                    AddCriticalMessage("* Quote Status is required")
                End If


                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

    End Sub

End Class

End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteContractsOption3/QuoteContractOption3/QuoteContractOption3Validator.cs
using System.Collections;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace QuoteContractOption3s
    {
        // make sure that contact object is valid
        public class QuoteContractOption3Validator : BaseValidator
        {
            public void Validate(QuoteContractOption3 oQuoteContractOption3)
            {
                // make sure that contact object is valid
                if (oQuoteContractOption3.Errors.Count > 0)
                {
                    foreach (DictionaryEntry de in oQuoteContractOption3.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (oQuoteContractOption3.QuoteContractReference.Trim().Length == 0)
                {
                    AddCriticalMessage("* Quote Contract Referenece is required");
                }

                if (Strings.InStr(oQuoteContractOption3.QuoteContractReference, "-") > 0)
                {
                    AddCriticalMessage("* Quote Contract Referenece cannot contain '-'.");
                }

                if (oQuoteContractOption3.QuoteContractStatusID == 0)
                {
                    AddCriticalMessage("* Quote Status is required");
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}